service-statistics-sotka/src/main/java/fi/nls/oskari/control/statistics/plugins/sotka/SotkaIndicatorValuesFetcher.java

```diff
@@ -6,105 +6,131 @@
 import java.util.Map.Entry;
 
+import fi.nls.oskari.control.statistics.data.IndicatorValueFloat;
+import fi.nls.oskari.control.statistics.plugins.sotka.requests.IndicatorDataJSON;
+import org.json.JSONArray;
 import org.json.JSONException;
 
 import fi.nls.oskari.control.statistics.plugins.APIException;
 import fi.nls.oskari.control.statistics.data.IndicatorValue;
 import fi.nls.oskari.control.statistics.data.StatisticalIndicatorDataDimension;
 import fi.nls.oskari.control.statistics.data.StatisticalIndicatorDataModel;
-import fi.nls.oskari.control.statistics.plugins.sotka.parser.SotkaIndicatorDataParser;
 import fi.nls.oskari.control.statistics.plugins.sotka.parser.SotkaRegionParser;
-import fi.nls.oskari.control.statistics.plugins.sotka.requests.IndicatorData;
 import fi.nls.oskari.control.statistics.plugins.sotka.requests.SotkaRequest;
+import org.json.JSONObject;
 
 /**
  * This fetches the indicator value tables transparently from Sotka.
  * We don't want to make a separate call to the plugin interface for this, because some
  * APIs / plugins might give all the information in the same response, or divide and key the responses differently.
  */
 public class SotkaIndicatorValuesFetcher {
-    private SotkaIndicatorDataParser parser;
     private SotkaRegionParser regionParser;
     private SotkaConfig config;
 
     public void init(SotkaConfig config) {
         this.config = config;
-        this.parser = new SotkaIndicatorDataParser();
         // We need to filter by region category and index by codes. Codes are unique within
-        if(this.regionParser == null) {
+        if (this.regionParser == null) {
             this.regionParser = new SotkaRegionParser(config);
         }
         regionParser.getData();
     }
 
+    /**
+     * @param selectors        Used to query SotkaNET with.
+     * @param indicator        The indicator we want.
+     * @param regionCategoryId The oskari layer we are interested in. For example: "KUNTA"
+     * @return
+     */
+    public Map<String, IndicatorValue> get(StatisticalIndicatorDataModel selectors, String indicator,
+                                           String regionCategoryId) {
+        Map<Integer, IndicatorValue> allValues = fetchData(selectors, indicator);
+        Map<String, IndicatorValue> filteredValues = new HashMap<>();
+        for (Entry<Integer, IndicatorValue> entry : allValues.entrySet()) {
+            Integer regionId = entry.getKey();
+            if (regionId == null) {
+                // sometimes this is null with 0.0 as value.
+                // TODO: check if this is a bug in some other code or an empty row from sotkanet
+                continue;
+            }
+
+            IndicatorValue value = entry.getValue();
+            // SotkaNET gives "Kunta" in some places, "KUNTA" in others...
+            String category = regionParser.getCategoryById(regionId);
+            if (category != null && category.equalsIgnoreCase(regionCategoryId)) {
```

Comment:
Could avoid the null check by swapping order, e.g. `if (regionCategoryId.equalsIgnoreCase(category))`